tests/QueryBuilderTest.php

```diff
@@ -109,7 +157,10 @@ public function it_keeps_local_macros_added_to_the_base_query()
 
         $queryBuilder = QueryBuilder::for($baseQuery);
 
-        $this->assertEquals($baseQuery->customMacro()->toSql(), $queryBuilder->customMacro()->toSql());
+        $this->assertEquals(
+            'select * from `test_models` where `name` = ?',
+            $queryBuilder->customMacro()->toSql()
+        );
```

Comment:
Now if we call a macro method both on `$baseQuery` and `$queryBuilder` it will result in duplication of `name` condition.
Because now the subject we pass to the `QueryBuilder` is mutable.